examples/Conversational_agents_using_tools.ipynb

```diff
@@ -0,0 +1,1095 @@
+     ]
+    }
+   ],
+   "source": [
+    "import os\n",
+    "import pinecone\n",
+    "import openai\n",
+    "import pandas as pd\n",
+    "import re\n",
+    "from typing import List, Union\n",
+    "import zipfile\n",
+    "import json\n",
+    "from tqdm.auto import tqdm\n",
+    "import datetime\n",
+    "\n",
+    "# Langchain imports\n",
+    "from langchain.agents import Tool, AgentExecutor, LLMSingleActionAgent, AgentOutputParser\n",
+    "from langchain.prompts import BaseChatPromptTemplate, ChatPromptTemplate\n",
+    "from langchain import SerpAPIWrapper, LLMChain\n",
+    "from langchain.schema import AgentAction, AgentFinish, HumanMessage, SystemMessage\n",
+    "# LLM wrapper\n",
+    "from langchain.chat_models import ChatOpenAI\n",
+    "from langchain import OpenAI\n",
+    "# Conversational memory\n",
+    "from langchain.memory import ConversationBufferWindowMemory\n",
+    "# Embeddings and vectorstore\n",
+    "from langchain.embeddings.openai import OpenAIEmbeddings\n",
+    "from langchain.vectorstores import Pinecone\n",
+    "\n",
+    "# Vectorstore Index\n",
+    "index_name = 'podcasts'"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "af825644",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "api_key = os.getenv(\"PINECONE_API_KEY\")\n",
+    "pinecone.init(api_key=api_key)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "c0518596",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "['podcasts']"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "pinecone.list_indexes()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7e7644cd",
+   "metadata": {},
+   "source": [
+    "Run this code block if you want to clear the index, or if the index doesn't exist yet\n",
+    "\n",
+    "```\n",
+    "# Check whether the index with the same name already exists - if so, delete it\n",
+    "if index_name in pinecone.list_indexes():\n",
+    "    pinecone.delete_index(index_name)\n",
+    "    \n",
+    "# Creates new index\n",
+    "pinecone.create_index(name=index_name, dimension=1536)\n",
+    "index = pinecone.Index(index_name=index_name)\n",
+    "\n",
+    "# Confirm our index was created\n",
+    "pinecone.list_indexes()\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0eb63e37",
+   "metadata": {},
+   "source": [
+    "## LLM Agent\n",
+    "\n",
+    "An [LLM agent](https://python.langchain.com/en/latest/modules/agents/agents/custom_llm_agent.html) in Langchain has many configurable components, which are detailed in the Langchain documentation.\n",
+    "\n",
+    "We'll employ a few of the core concepts to make an agent who talks in the way we want, can use tools to answer questions, and uses the right base LLM to power the conversation.\n",
```

Comment:
Maybe just "language model" or "model" instead of "base LLM"?

- Generally I try to avoid abbreviations
- I'm also not sure what 'base' means here